ykrt/build.rs

```diff
@@ -1,6 +1,59 @@
+use regex::Regex;
 use std::env;
+use std::os::unix::fs::symlink;
+use std::path::PathBuf;
+use std::process::Command;
+
+include!("../build_aux.rs");
+
+const YKSHIM_SO: &str = "libykshim.so";
 
 fn main() {
+    let ykrt_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    let mut internal_dir = ykrt_dir.clone();
+    internal_dir.push("..");
+    internal_dir.push("internal_ws");
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let cargo = env::var("CARGO").unwrap();
+    let rust_flags = env::var("RUSTFLAGS").unwrap_or_else(|_| String::new());
+    let tracing_kind = find_tracing_kind(&rust_flags);
+
+    // Build the internal workspace.
+    env::set_current_dir(&internal_dir).unwrap();
```

Comment:
Always running `cargo build` inside `build.rs` will consume the output of the build process unless it fails, so you won't see any warnings, nor will you get any progress indication. Maybe `build.rs` could take up the responsibility only if you aren't using `cargo xtask`?

Reply:
Is there a way to know if we are using xtask?

I also tried having `build.rs` run `xtask` but it deadlocks, as the two cargos fight for the lock file :(

Reply:
xtask could set an env var.

Reply:
We could make warnings become errors in build.rs.

Reply:
That would still not give a progress indication. In addition during development it is often useful to have warnings be warnings while still seeing them.

Reply:
Why is the output different actually? Does rustc know that it's attached to a pipe, or?

Reply:
> xtask could set an env var.

c8823b7c0 implements this. Seems to work.